### DISCUSSION TITLE
Hooking into compilation of scanner.c

### DISCUSSION THREAD

**Discussion by asvarga** [tree-sitter/tree-sitter#2604] [General]
Is there any way to hook into tree-sitter's compilation of the `scanner.c` file to do additional compilation steps? I want to use some code that I can't just `#include` or modify, but would be able to use if I could compile it.

**Comment by ahlinc** [comment]
No, there is no way to hook how Tree-sitter's CLI compiles scanners' code and due to our plans to implement dependency free scanners we don't plan to implement any of such hooks. For improving Tree-sitter grammars portability for WASM limited runtimes and others, we plan to introduce more restrictive rules for scanners and issue a guide of how to write scanners even without libc dependencies.

You can only achieve what you want by using your own compilation process outside the CLI Tree-sitter for a grammar's parser and scanner code.
